flake.nix

```diff
@@ -0,0 +1,78 @@
+{
+  description = "tkn-watch - watch a PipelineRuns on its way to success or failures";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    naersk.url = "github:nmattia/naersk";
+    naersk.inputs.nixpkgs.follows = "nixpkgs";
+  };
+
+  outputs = { self, nixpkgs, naersk }:
+    let
+      cargoToml = (builtins.fromTOML (builtins.readFile ./Cargo.toml));
+      supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" ];
```

Comment:
i had issues with cross compilation on different host with the kube-rs lib (openssl) but i believe it's fixed so you can maybe add aarch64-darwin (if that's the name)